lib/pact/support/metrics.rb

```diff
@@ -0,0 +1,85 @@
+require 'securerandom'
+require 'digest'
+require 'socket'
+require 'pact/support/metrics'
+require 'pact/mock_service/version'
+require 'net/http'
+
+module Pact
+  module Support
+    class Metrics
+
+      def self.report_metric(event, category, action, value = 1)
+        in_thread do
+          begin
+            if track_events?
+                Pact.configuration.output_stream.puts "WARN: Please note: we are tracking events anonymously to gather important usage statistics like Pact-Ruby version and operating system. To disable tracking, set the 'PACT_DO_NOT_TRACK' environment variable to 'true'."
+
+                uri = URI('https://www.google-analytics.com/collect')
+                req = Net::HTTP::Post.new(uri)
+                req.set_form_data(create_tracking_event(event, category, action, value))
+
+                Net::HTTP.start(uri.hostname, uri.port, read_timeout:2, open_timeout:2, :use_ssl => true  ) do |http|
+                  http.request(req)
+              end
+            end
+          rescue StandardError => e
+            handle_error(e)
+          end
+        end
+      end
+
+      private
+
+      def self.handle_error e
+        if ENV['PACT_METRICS_DEBUG'] == 'true'
+          Pact.configuration.output_stream.puts("DEBUG: #{e.inspect}")
+        end
```

Comment:
I just did a quick test on e.inspect, and it doesn't give a stack trace. Can you also print the backtrace please.

Reply:
This uses the standard ruby logger, which doesn't take 2 arguments.

```
$ irb
irb(main):001:0> require 'logger'
=> true
irb(main):002:0> l = Logger.new($stdout)
=> #<Logger:0x00007ff88d030c90 @level=0, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007ff88d030c68 @datetime...
irb(main):003:0> l.info("foo", "bar")
Traceback (most recent call last):
        5: from /redacted/bin/irb:23:in `<main>'
        4: from /redacted/bin/irb:23:in `load'
        3: from /redacted/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        2: from (irb):3
        1: from /redacted/lib/ruby/2.7.0/logger.rb:527:in `info'
ArgumentError (wrong number of arguments (given 2, expected 0..1))
```

You'll need to concatenate it like `("DEBUG: #{e.inspect}\n" + e.backtrace.join("\n))`